pyogrio/geopandas.py

```diff
@@ -309,9 +309,11 @@ def write_dataframe(
     # Determine geometry_type and/or promote_to_multi
     if geometry_type is None or promote_to_multi is None:
         tmp_geometry_type = "Unknown"
+        has_z = False
 
         # If there is data, infer layer geometry type + promote_to_multi
         if not df.empty:
+            has_z = geometry.has_z.any()
```

Comment:
We might want to filter out empties before calling `has_z` (something like `geometry[~geometry.is_empty].has_z.any()`). Depending on how the empty geometry is created, they tend to incorrectly think they are 3D (also depending on the GEOS version)

(I did something similar in shapely where we needed to determine to take a 2D vs 3D code path: https://github.com/shapely/shapely/blob/1ee6e96a4c84d1b99152b16ac96b71d2b8012c97/shapely/_ragged_array.py#L249-L252)

Reply:
Done